_pydevd_sys_monitoring/_pydevd_sys_monitoring.py

```diff
@@ -276,19 +276,15 @@ def __init__(self, thread: threading.Thread, thread_ident: int, trace: bool, add
         self.thread_ident = thread_ident
         self.additional_info = additional_info
         self.trace = trace
-        self._use_is_stopped = hasattr(thread, '_is_stopped')
-        
+
     # fmt: off
     # IFDEF CYTHON
     # cdef bint is_thread_alive(self):
     # ELSE
     def is_thread_alive(self):
     # ENDIF
     # fmt: on
-        if self._use_is_stopped:
-            return not self.thread._is_stopped
-        else:
-            return not self.thread._handle.is_done()
+        return is_thread_alive(self.thread)
```

Comment:
Ideally any checks for Python 3.12 onwards would be done here and not delegated to a pure-python implementation (i.e.: the constructor should determine which check to do and here it uses the proper approach without calling out to non-cython code).

Reply:
Roger that, fixed